File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def bfs(i, computers, visited):
    q = deque([i])

    while q:
        item = q.popleft()
        visited[item] = True
        
        for n in range(len(computers)):
            if not visited[n] and computers[item][n] == 1:
                q.append(n)

def solution(n, computers):
    answer = 0

    visited = [False] * n
    for i in range(n):
        if not visited[i]:
            bfs(i, computers, visited)
            answer += 1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    void bfs(int start, int[][] computers, boolean[] visited) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        visited[start] = true;
        
        while (!q.isEmpty()) {
            int item = q.poll();
            for (int i = 0; i < computers[item].length; i++) {
                if (computers[item][i] != 0 && visited[i] != true) {
                    visited[i] = true;
                    q.offer(i);
                }
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (visited[i] != true) {
                bfs(i, computers, visited);
                answer += 1;
            }
        }
        return answer;
    }
}